gateway-provider-security-jwt/src/main/java/org/apache/knox/gateway/provider/federation/jwt/filter/AbstractJWTFilter.java

```diff
@@ -187,6 +186,22 @@ protected boolean tokenIsStillValid(JWT jwtToken) throws UnknownTokenException {
     return expires == null || new Date().before(expires);
   }
 
+  protected boolean tokenIsStillValid(final String tokenId) throws UnknownTokenException {
```

Comment:
Is it possible here that an unknown token will return null and be considered still valid?

Reply:
I don't believe so; The UnknownTokenException should work its way back to the validateToken method where it is handled.

Reply:
Make sure we have a test for bogus tokens to be sure.

Reply:
I'm pretty sure TokenIDAsHTTPBasicCredsFederationFilterTest#testInvalidJWTForPasscode() verifies this.